[BUG]: Clippy Check fails due to build issue on Linux

### Description

When running `clippy` via GitHub action, the check fails due to build error. This may require adding or updating dependencies for Ubuntu runner.

### Steps to Reproduce

**Assume the project is formatted and linted**
1. Make PR to `main` branch
2. Wait for `backend-check` job to complete
3. It describes the error in the clippy check step

### Expected Behavior

Show same clippy message as on user's computer when build works.

### Logs

```shell
cd ./src-tauri; cargo clippy -- -D warnings
# ...
error: failed to run custom build command for `alsa-sys v0.3.1`

Caused by:
  process didn't exit successfully: `/home/runner/work/SuperMouseAI/SuperMouseAI/src-tauri/target/debug/build/alsa-sys-5b1d766648e03ecb/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=ALSA_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=ALSA_STATIC
  cargo:rerun-if-env-changed=ALSA_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr

  thread 'main' panicked at /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/alsa-sys-0.3.1/build.rs:13:18:

  pkg-config exited with status code 1
  > PKG_CONFIG_ALLOW_SYSTEM_LIBS=1 PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 pkg-config --libs --cflags alsa

  The system library `alsa` required by crate `alsa-sys` was not found.
  The file `alsa.pc` needs to be installed and the PKG_CONFIG_PATH environment variable must contain its parent directory.
  The PKG_CONFIG_PATH environment variable is not set.

  HINT: if you have installed the library, try setting PKG_CONFIG_PATH to the directory containing `alsa.pc`.

  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.
```

### Environment

OS: Ubuntu

### Additional Context

_No response_

--- a/.github/workflows/rust-check.yml
+++ b/.github/workflows/rust-check.yml
@@ -25,6 +25,7 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get install -y libwebkit2gtk-4.1-dev libappindicator3-dev librsvg2-dev patchelf libasound2-dev
+          touch ./src-tauri/resources/whisper-model.bin
 
       - name: Run Rust Lints
         run: cd ./src-tauri; cargo clippy -- -D warnings

--- a/.github/workflows/rust-check.yml
+++ b/.github/workflows/rust-check.yml
@@ -24,7 +24,7 @@ jobs:
       - name: Install Dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install -y libwebkit2gtk-4.1-dev libappindicator3-dev librsvg2-dev patchelf
+          sudo apt-get install -y libwebkit2gtk-4.1-dev libappindicator3-dev librsvg2-dev patchelf libasound2-dev
 
       - name: Run Rust Lints
         run: cd ./src-tauri; cargo clippy -- -D warnings

--- a/.github/workflows/rust-check.yml
+++ b/.github/workflows/rust-check.yml
@@ -28,4 +28,4 @@ jobs:
           touch ./src-tauri/resources/whisper-model.bin
 
       - name: Run Rust Lints
-        run: cd ./src-tauri; cargo clippy -- -D warnings
+        run: cd ./src-tauri; cargo clippy --all-targets -- -D warnings

--- a/.github/workflows/rust-check.yml
+++ b/.github/workflows/rust-check.yml
@@ -20,5 +20,11 @@ jobs:
       - name: Check Rust Formatting
         run: cd ./src-tauri; cargo fmt --all -- --check
 
+      # Install 
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libwebkit2gtk-4.1-dev libappindicator3-dev librsvg2-dev patchelf
+
       - name: Run Rust Lints
         run: cd ./src-tauri; cargo clippy -- -D warnings

--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -51,6 +51,9 @@ whisper-rs = { version = "0.14.2", features = [] }
 [target.'cfg(target_os = "macos")'.dependencies]
 whisper-rs = { version = "0.14.2", features = ["metal", "coreml"] }
 
+[target.'cfg(target_os = "linux")'.dependencies]
+whisper-rs = { version = "0.14.2", features = [] }
+
 [features]
 default = ["symphonia-all", "minimp3", "openmp"]
 

--- a/src-tauri/src/lib.rs
+++ b/src-tauri/src/lib.rs
@@ -7,6 +7,7 @@
 use log::{debug, error, info, trace, warn, LevelFilter};
 use std::{collections::HashMap, path::PathBuf};
 use tauri::{path::BaseDirectory, Manager};
+#[allow(unused_imports)]
 use tauri_plugin_sentry::{minidump, sentry};
 use tauri_specta::{Builder, Event};
 

